torchaudio/prototype/ctc_decoder/ctc_decoder.py

```diff
@@ -177,7 +177,7 @@ def kenlm_lexicon_decoder(
 
     Args:
         lexicon (str): lexicon file containing the possible words
-        tokens (str): file containing valid tokens
+        tokens (str or List[str]): file or list containing valid tokens
```

Comment:
not in the scope of the pr, but is there some reference that we can cite to help users understand how tokens files should be formatted?

Reply:
good point, this is somewhat shown in the [tutorial](https://pytorch.org/audio/main/tutorials/asr_inference_with_ctc_decoder_tutorial.html#tokens) but could definitely be improved. I'm thinking either in some README section (w/ docstrings linking to it), directly in the docstrings, or in more detail in a tutorial if you have any thoughts?

Reply:
I think we can get rid of loading from the file-path.
`List[str]` is simple to construct, users can format data such way after loading data from a file in a different format.
I am looking at the changes to tests, but keeping the pass-like type increase the test cost.
Specifically, I think `test_construct_decoder` is getting redundant yet not enough without `test_index_to_tokens`.

Reply:
there is a case from flashlight [README](https://github.com/flashlight/flashlight/blob/9193b92e86538128664c1cf22785f4f5778c8312/flashlight/app/asr/README.md#token-dictionary) that can not be done using `List[str]`, where they mention `If two tokens are on the same line in the tokens file, they are mapped to the same index for training/decoding`. I'm not sure of specific use cases of this but may be good to have the option open?

Reply:
Huh🤔 so in the case of ASR, this is to allow one phoneme to be mapped to different tokens. 

Okay, then let's keep it.

Reply:
> I'm thinking either in some README section (w/ docstrings linking to it), directly in the docstrings, or in more detail in a tutorial if you have any thoughts?

yeah that all seems reasonable. i think if the format is simple enough to describe concisely, it'd be ideal to include it directly in the docstrings for ease of access. otherwise, linking to a readme sounds good

re: the file loading — i guess if the file format is standardized and something users are accustomed to, it'd be helpful to allow for directly loading files of that format. otherwise, it seems like pulling out the file loading logic would yield a more minimalist/flexible solution. the one-index-to-many-tokens case wouldn't necessarily require file loading — we could account for it with a list of lists or dictionary input